packages/dds/map/src/directory.ts

```diff
@@ -1478,7 +1480,9 @@ class SubDirectory extends TypedEventEmitter<IDirectoryEvents> implements IDirec
 			dirs: this._subdirectories,
 			next(): IteratorResult<[string, IDirectory]> {
 				if (this.index < subdirNames.length) {
-					const subdirName = subdirNames[this.index++];
+					// TODO Non null asserting, why is this not null?
+					// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+					const subdirName = subdirNames[this.index++]!;
```

Comment:
This is the sort of reason why I hate increment operators inline with access, I always split the increment to a separate line when I see it.  Just more mental overhead when reading the code.

```suggestion
					// Non null asserting, we've checked that the index is inside the bounds of the array.
					// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
					const subdirName = subdirNames[this.index]!;
					this.index++;
```

Reply:
I agree